mirroring/handler.go

```diff
@@ -0,0 +1,230 @@
+package mirroring
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/go-openapi/strfmt"
+	trilliantypes "github.com/google/trillian/types"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spf13/viper"
+)
+
+// consider adding a rekorClient field so that the same client struct is used for all operations, potentially saving some time
+type LogHandler struct {
+	metadata        TreeMetadata
+	client          *client.Rekor
+	newLeavesBuffer []Artifact
+}
+
+func LoadFromRemote(serverURL string) (h LogHandler, err error) {
+	viper.Set("rekorServerURL", serverURL)
+	viper.SetDefault("metadata_file_dir", "./.newmetadata")
+	viper.SetDefault("tree_file_dir", "./.newtree")
+	pub, err := GetPublicKey()
+	if err != nil {
+		return
+	}
+
+	logInfo, err := GetLogInfo()
+	if err != nil {
+		return
+	}
+
+	metadata := TreeMetadata{PublicKey: pub, LogInfo: logInfo, SavedMaxIndex: -1}
+	h.metadata = metadata
+
+	h.client, err = NewClient()
+	if err != nil {
+		return h, err
+	}
+	return
+}
+
+// LoadFromLocal parses a JSON file to create a log handler that can be used
+// to fetch and verify properties about the log.
+func LoadFromLocal(fileName string) (LogHandler, error) {
+	viper.SetConfigFile(fileName)
+	viper.AddConfigPath(".")
+	err := viper.ReadInConfig() // Find and read the config file
+	if err != nil {
+		panic(fmt.Errorf("configuration file could not be found.\n%s", err))
+	}
+
+	handler := LogHandler{}
+	metadata, err := LoadTreeMetadata()
+	if err != nil {
+		return handler, err
+	}
+	handler.client, err = NewClient()
+	if err != nil {
+		return handler, err
+	}
+	handler.metadata = metadata
+	// stored information will contain duplicates. should these duplicates be checked?
```

Comment:
Can we ticketize this?

Reply:
I removed the comment and added the checks in the next commit.